pkg/kubelet/eviction/types.go

```diff
@@ -132,19 +132,22 @@ type nodeReclaimFunc func() error
 type nodeReclaimFuncs []nodeReclaimFunc
 
 // thresholdNotifierHandlerFunc is a function that takes action in response to a crossed threshold
-type thresholdNotifierHandlerFunc func(thresholdDescription string)
-
-// ThresholdStopCh is an interface for a channel which is closed to stop waiting goroutines.
-// Implementations of ThresholdStopCh must correctly handle concurrent calls to all functions.
-type ThresholdStopCh interface {
-	// Reset closes the channel if it can be closed, and returns true if it was closed.
-	// Reset also creates a new channel.
-	Reset() bool
-	// Ch returns the channel that is closed when Reset() is called
-	Ch() <-chan struct{}
-}
+type thresholdNotifierHandlerFunc func(message string)
 
-// ThresholdNotifier notifies the user when an attribute crosses a threshold value
+// ThresholdNotifier performs a function when a cgroup attribute crosses a threshold value
 type ThresholdNotifier interface {
-	Start(ThresholdStopCh)
+	// UpdateThreshold removes the previous threshold and adds the specified threshold to the specified attribute
+	// in the cgroup specified by the path.
+	UpdateThreshold(path, attribute, threshold string) error
+}
+
+// MemoryThresholdNotifier manages ThresholdNotifiers based on memory eviction thresholds, and performs a function
+// when memory eviction thresholds are crossed
+type MemoryThresholdNotifier interface {
```

Comment:
Maybe a better name is `NotifierManager` or something.  I wanted to separate out the management of thresholds (I.E. taking eviction concepts like thresholds and the summary API and turning them into a set of kernel changes (using cgroup control, and managing event file descriptors)

Reply:
Let me know what you think, and ill update it

Reply:
Having 2 interfaces (one that integrates with cgroups, and one that configures it specifically for the memory eviction thresholds) allows me to mock the one that does all of the integration with linux, and do unit tests on the one that does the configuration based on the eviction thresholds.

I am not sure the ThresholdNotifierSet would actually make the eviction manager code simpler.  It would just mean removing a single for loop from `synchronize`.  Or maybe I am missing something.